burn-wgpu/src/ops/base.rs

```diff
@@ -209,4 +237,139 @@ impl<G: GraphicsApi> BaseOps<G> {
 
         tensor
     }
+
+    pub fn equal<E: WgpuElement, const D: usize>(
+        lhs: WgpuTensor<E, D>,
+        rhs: WgpuTensor<E, D>,
+    ) -> WgpuTensor<u32, D> {
+        let can_be_used_as_bool = mem::size_of::<E>() == mem::size_of::<u32>();
+
+        if can_be_used_as_bool && lhs.can_mut_broadcast(&rhs) {
+            return comparison_inplace::<EqualInplace, E, D>(lhs, rhs);
+        }
+        if can_be_used_as_bool && rhs.can_mut_broadcast(&lhs) {
+            return comparison_inplace::<EqualInplace, E, D>(rhs, lhs);
+        }
+
+        comparison::<Equal, E, D>(lhs, rhs)
+    }
+
+    pub fn greater<E: WgpuElement, const D: usize>(
+        lhs: WgpuTensor<E, D>,
+        rhs: WgpuTensor<E, D>,
+    ) -> WgpuTensor<u32, D> {
+        let can_be_used_as_bool = mem::size_of::<E>() == mem::size_of::<u32>();
+
+        if can_be_used_as_bool && lhs.can_mut_broadcast(&rhs) {
+            return comparison_inplace::<GreaterInplace, E, D>(lhs, rhs);
+        }
+        if can_be_used_as_bool && rhs.can_mut_broadcast(&lhs) {
+            return comparison_inplace::<LowerInplace, E, D>(rhs, lhs);
+        }
+
+        comparison::<Greater, E, D>(lhs, rhs)
+    }
+
```

Comment:
Do we need documentation for these pub methods here? Or these implementing a trait? Or these are pub (crate) methods?

Reply:
Everything is pretty much pub(crate) except the device and the backend struct.